repose-aggregator/components/services/datastore-service/datastore-service-impl/src/test/scala/org/openrepose/nodeservice/distributed/servlet/DistributedDatastoreServletTest.scala

```diff
@@ -0,0 +1,93 @@
+import javax.servlet.http.HttpServletResponse._
+
+import io.opentracing.Tracer
+import io.opentracing.mock.MockTracer
+import org.junit.runner.RunWith
+import org.mockito.Matchers.anyString
+import org.mockito.Mockito
+import org.mockito.Mockito.verify
+import org.openrepose.core.services.datastore.distributed.ClusterConfiguration
+import org.openrepose.core.services.datastore.distributed.config._
+import org.openrepose.core.services.datastore.impl.distributed.CacheRequest.CACHE_URI_PATH
+import org.openrepose.core.services.datastore.{Datastore, DatastoreAccessControl, DatastoreService}
+import org.scalatest.junit.JUnitRunner
+import org.scalatest.mock.MockitoSugar
+import org.scalatest.{BeforeAndAfterEach, FunSpec, Matchers}
+import org.springframework.mock.web.{MockHttpServletRequest, MockHttpServletResponse}
+
+
+@RunWith(classOf[JUnitRunner])
+class DistributedDatastoreServletTest extends FunSpec with BeforeAndAfterEach with Matchers with MockitoSugar {
+
+  private val mockDatastoreService = mock[DatastoreService]
+  private val mockDatastore = mock[Datastore]
+  Mockito.when(mockDatastoreService.getDefaultDatastore).thenReturn(mockDatastore)
+  private val distributedDatastoreConfiguration = mock[DistributedDatastoreConfiguration]
+  val tracerSpy = Mockito.spy(new MockTracer)
+
+  val distributedDatastoreServlet = new DistributedDatastoreServlet(
+    mockDatastoreService,
+    mock[ClusterConfiguration],
+    new DatastoreAccessControl(Collections.emptyList[InetAddress], true),
+    distributedDatastoreConfiguration,
+    tracerSpy
+  )
+
+  var servletRequest: MockHttpServletRequest = _
+  var servletResponse: HttpServletResponse = _
+
+
+  override def beforeEach(): Unit = {
+    servletRequest = new MockHttpServletRequest
+    servletRequest.setRequestURI(CACHE_URI_PATH)
+    servletRequest.setProtocol("1.1")
+    servletResponse = new MockHttpServletResponse
+    tracerSpy.reset()
+  }
+
+  describe("Distributed Datastore calls without a Cache Key") {
+    val notFound = List("GET", "HEAD").map(method => (method, SC_NOT_FOUND)) // Head calls Get and no Cache Key sent
+    val badRequest = List("PUT", "PATCH").map(method => (method, SC_BAD_REQUEST)) // No data or Cache Key sent
+    val notAllowed = List("POST", "TRACE").map(method => (method, SC_METHOD_NOT_ALLOWED)) // Methods Not Allowed
+    val okMethods = List("OPTIONS").map(method => (method, SC_OK)) // This is handled by the parent HTTP Servlet
+    val notImplemented = List("CUSTOM", "BOGUS").map(method => (method, SC_NOT_IMPLEMENTED)) // Not Implemented
+
+    (notFound ++ badRequest ++ notAllowed ++ okMethods ++ notImplemented) foreach { case (httpMethod, result) =>
+      it(s"should build an OpenTracing span for HTTP method $httpMethod") {
+        servletRequest.setMethod(httpMethod)
+        distributedDatastoreServlet.service(servletRequest, servletResponse)
+        verify(tracerSpy).buildSpan(s"$httpMethod $CACHE_URI_PATH")
```

Comment:
I'm not sure you actually need to use a `spy` here. Wouldn't `MockTracer.finishedSpans()` give you the same information since `service` is a synchronous call? So in this case, couldn't you replace this line with:
```
tracer.finishedSpans.size shouldEqual 1
tracer.finishedSpans.get(0).operationName() shouldEqual s"$httpMethod $CACHE_URI_PATH"
```

Of course, I have nothing against spies, I just found it interesting that a `spy` was being used in conjunction with a `MockTracer` since they are serving a similar purpose.

Reply:
Yep. I hadn't fully explored the `MockTracer` API and knew the `spy` would work.

Updated.